### DISCUSSION TITLE
Error Returned During Cluster Init

### DISCUSSION THREAD

**Discussion by HomeLabRedneck** [techno-tim/k3s-ansible#250] [Questions] [answered]
I am deploying on 5 ubuntu 22.04 VMs. Everything up to this error deploys successfully. I am guessing the error during the verification is occurring because of the error during init.

![image](https://user-images.githubusercontent.com/127556886/224451360-656203fe-2bb4-4486-8a2a-1f491fbb1b31.png)

**Comment by timothystewart6** [comment, marked as answer]
See https://github.com/techno-tim/k3s-ansible/discussions/20    1)

also, this will help you get latest https://docs.technotim.live/posts/ansible-automation/#installing-the-latest-version-of-ansible

**Comment by HomeLabRedneck** [comment]
I am on the newest version of Ansible.
![image](https://user-images.githubusercontent.com/127556886/224710685-ff7c8018-820d-457e-b233-4d10b8a00ab3.png)

✅ Installed and verified that ansible requirements are installed with ansible-galaxy install -r ./collections/requirements.yml
![image](https://user-images.githubusercontent.com/127556886/224711140-b4311132-ce27-403c-b500-43ebc72c831c.png)

✅ Variables match the ones used below, including no additional args.
I have tried with no additional variables and then a couple of minor variables such as:
--write-kubeconfig-mode 644
--kublet-arg node-status-update-frequency=5s

✅ That all server / master nodes have the same network interface name (flannel_iface this cannot vary)
All nodes master and nodes have the same interface name: enp0s3

✅ If they do not match, create one master / server node and add additional servers outside of this playbook
N/A

✅ All machines have enough disk space, at least 4 GB (minimum)
All VMs have 60GB

✅ Drives are fast enough to run etcd (SSD or faster)
SSD storage

✅ Network / disk latency is low enough to run etcd
All master and nodes are running on the same host

✅ Run the reset command, ensure this reboots all machines.
![image](https://user-images.githubusercontent.com/127556886/224713085-ca892dbf-bdf9-4c95-ba33-b4fd5a02eab8.png)

✅ Same time zone on all machines
Time zone is set by the playbook
![image](https://user-images.githubusercontent.com/127556886/224713538-eece945c-251a-4614-a0bd-51a477116d8f.png)

✅ k3s_token does not contain special characters (alpha numeric is preferred)
It does not contain special character, alpha numeric only.

✅ ssh keys work on all nodes
SSH keys work on all nodes

✅ ansible_user is the user that has ssh access via ansible
I didn't not use ansible_user, I used my own username but did a find and replace with vscode for all files.

**Comment by timothystewart6** [comment]
Can you run a reset and try again?  The first failure is due to ansible version.  Split was not available until `2.11`.

**Reply by HomeLabRedneck** [reply]
Will do.

**Reply by HomeLabRedneck** [reply]
![image](https://user-images.githubusercontent.com/127556886/224805292-210bab5d-f263-4fc0-80ff-7713ab1717de.png)

Same error as before. I forgot to mention that before I could ping the VIP and I still can.
![image](https://user-images.githubusercontent.com/127556886/224806322-4f2201b6-6b07-4627-a9cb-379c03c91a45.png)

scp serveradmin@192.168.70.5:~/.kube/config ~/.kube/config  -- This command returns nothing. 

kubectl is not recognized as a command on any of the VMs.

**Reply by HomeLabRedneck** [reply]
This is executed on the primary master node.
![image](https://user-images.githubusercontent.com/127556886/224807612-f92fa30d-8604-43b3-acc1-dac8e1b6cb38.png)

**Reply by timothystewart6** [reply]
If you are getting the split error, nothing else will work.  This is typically solved by installing the latest version of ansible as [shown here](https://docs.technotim.live/posts/ansible-automation/#installing-the-latest-version-of-ansible)  this is definitely something with your local environment because we build a cluster in CI and that works https://github.com/techno-tim/k3s-ansible/actions/runs/4339976288

You'll also want to be sure you run the reset playbook to be sure that all nodes are cleaned up

**Reply by HomeLabRedneck** [reply]
Does the split command attempt to run on all the master nodes? Which means that it was successful on the primary master if that is the case. 

Cloud cloning the first master to create 2 more masters be causing the issue?

**Reply by HomeLabRedneck** [reply]
I created new VMs right from the iso. so that is not the issue. Not sure why it would work on one VM but not the others.

**Reply by HomeLabRedneck** [reply]
I figured it out, your troubleshooting guide was correct in identifying the issue. What I didn't do the first time was sudo su into the serveradmin account, which had an earlier version of ansible. Thank you for your help Tim.

**Reply by timothystewart6** [reply]
No problem!  Glad you figured it out!
